src/Symfony/Bridge/Doctrine/Listener/DoctrineConnectionListener.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Doctrine\Listener;
+
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+class DoctrineConnectionListener implements EventSubscriberInterface
+{
+    public function __construct(private readonly \ArrayObject $connectionTiming, private ContainerInterface $container, private readonly array $timingbyConnection)
+    {
+    }
+
+    public function onKernelRequest(RequestEvent $event): void
+    {
+        $connectionTiming = $this->connectionTiming->getIterator();
+
+        if (!$connectionTiming) {
+            return;
+        }
+
+        foreach ($connectionTiming as $key => $value) {
+            $connectionName = sprintf('doctrine.dbal.%s_connection', $value['name']);
+
+            if (time() - $value['timing'] >= $this->timingbyConnection[$connectionName]) {
+                $connection = $this->container->get(sprintf('doctrine.dbal.%s_connection', $value['name']));
```

Comment:
I suggest injecting the ConnectionRegistry (which allows getting connections by name) or a service locator indexed by connection name instead of hardcoding the service naming convention of DoctrineBundle in this bridge class.

Reply:
(not for other reviewers: this suggestion has been objected - using public service ids will provide the best performance)

Reply:
Even if we inject a ServiceLocator, we could still index it by connection name rather than service ids (it makes no difference in the performance of a ServiceLocator)

Reply:
We could, but there is zero benefit doing so: since connections are public services, we can skip building any tailored locator.